hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/ClientConfigXmlGenerator.java

```diff
@@ -0,0 +1,696 @@
+            appendAttributes(xml, attributes);
+            xml.append('>');
+        }
+
+        private static void appendCloseNode(StringBuilder xml, String name) {
+            xml.append("</").append(name).append('>');
+        }
+
+        private static void appendNode(StringBuilder xml, String name, Object contents, Object... attributes) {
+            if (contents != null || attributes.length > 0) {
+                xml.append('<').append(name);
+                appendAttributes(xml, attributes);
+                if (contents != null) {
+                    xml.append('>');
+                    escapeXml(contents, xml);
+                    xml.append("</").append(name).append('>');
+                } else {
+                    xml.append("/>");
+                }
+            }
+        }
+
+        private static void appendAttributes(StringBuilder xml, Object... attributes) {
+            for (int i = 0; i < attributes.length; ) {
+                xml.append(" ").append(attributes[i++]).append("=\"");
+                escapeXmlAttr(attributes[i++], xml);
+                xml.append("\"");
+            }
+        }
+
+        /**
+         * Escapes special characters in XML element contents and appends the result to <code>appendTo</code>.
+         */
+        private static void escapeXml(Object o, StringBuilder appendTo) {
+            if (o == null) {
+                appendTo.append("null");
+                return;
+            }
+            String s = o.toString();
+            int length = s.length();
+            appendTo.ensureCapacity(appendTo.length() + length + CAPACITY);
+            for (int i = 0; i < length; i++) {
+                char ch = s.charAt(i);
+                if (ch == '<') {
+                    appendTo.append("&lt;");
+                } else if (ch == '&') {
+                    appendTo.append("&amp;");
+                } else {
+                    appendTo.append(ch);
+                }
+            }
+        }
+
+        /**
+         * Escapes special characters in XML attribute value and appends the result to <code>appendTo</code>.
+         */
+        private static void escapeXmlAttr(Object o, StringBuilder appendTo) {
+            if (o == null) {
+                appendTo.append("null");
+                return;
+            }
+            String s = o.toString();
+            int length = s.length();
+            appendTo.ensureCapacity(appendTo.length() + length + CAPACITY);
+            for (int i = 0; i < length; i++) {
+                char ch = s.charAt(i);
+                switch (ch) {
+                    case '"':
+                        appendTo.append("&quot;");
+                        break;
+                    case '\'':
+                        appendTo.append("&#39;");
+                        break;
+                    case '&':
+                        appendTo.append("&amp;");
+                        break;
+                    case '<':
```

Comment:
technically you don't need to escape `<>` in attributes I think

Reply:
You need to escape `<`, you never need to escape `>`